src/pages/index.js

```diff
@@ -41,62 +182,136 @@ editProfileMOB.addEventListener("click", () => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
 }
 
+//Handles new card submit
 function handleAddCardSubmit(cardData) {
-  createCard(cardData, "prepend");
-  addPlacePopup.popupForm.reset();
-  addPlaceFormValidator.toggleButtonState();
-  addPlacePopup.close();
+  const submitButton = addCardModal.querySelector(
+    "#add-card-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardData)
+    .then((res) => {
+      addNewCard(res, "prepend");
+      addPlacePopup.popupForm.reset();
+      addPlaceFormValidator.toggleButtonState();
+      addPlacePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Create";
+    });
 }
 
-const handleImageClick = (cardData) => {
-  maxImagePopup.open(cardData);
-};
+//Handles card deletion
+let cardtoDeleteData;
 
-//***************************************************************************************************************************//
-//                                                     Objects                                                               //
-//***************************************************************************************************************************//
+//Handles confirm delete and sets delete data
+function handleConfirmDelete(data) {
+  confirmDeletePopup.open();
+  cardtoDeleteData = data;
+}
 
-const cardSection = new Section(initialCards, createCard, "#cards-list");
-const maxImagePopup = new PopupWithImage("#max-image-modal");
-const editProfilePopup = new PopupWithForm(
-  "#edit-profile-modal",
-  handleEditProfile
-);
-const addPlacePopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+//Deletes card
+async function handleDeleteCard() {
+  try {
+    api.deleteCard(cardtoDeleteData._id);
+    const card = document.querySelector(`li[id='${cardtoDeleteData._id}'`);
+    card.remove();
+  } catch (err) {
+    console.log(err);
+  } finally {
+    confirmDeletePopup.close();
+  }
+}
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__name",
-  jobSelector: ".profile__subtitle",
-});
+//Changes Profile Picture
+async function handleEditPicSubmit({ link }) {
+  const submitButton = editProfilePicModal.querySelector(
+    "#edit-pic-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
 
-//***************************************************************************************************************************//
-//                                             Validation and Rendering                                                      //
-//***************************************************************************************************************************//
+  try {
+    api.changeProfilePicture(link);
+  } catch (err) {
+    console.log(err);
+  } finally {
+    profilePic.src = link;
+    profilePicPopup.close();
```

Comment:
* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well